hummingbot/market/bamboo_relay/bamboo_relay_market.pyx

```diff
@@ -475,8 +487,17 @@ cdef class BambooRelayMarket(MarketBase):
                     locked_balances[currency] = locked_balances.get(currency, s_decimal_0) + amount
 
                 for currency, balance in total_balances.items():
-                    self._account_available_balances[currency] = \
-                        Decimal(total_balances[currency]) - locked_balances.get(currency, s_decimal_0)
+                    asset_limit = exchange_limits.get(currency.upper(), None)
```

Comment:
It'd be better if this part is generalised, ie. have a function called `apply_balance_restriction` in `market_base` class. For all subclasses that need to do this can call the function after update its balance (from API).

Reply:
Will do